### ISSUE TITLE
Support other languages

### ISSUE THREAD

**Issue by jayvdb** [DanielVenturini/vigilant-lamp#3]
It would be useful to have this type of functionality for other languages/package managers too.

pip is the most obvious candidate, given this is a Python library.

I think another which would be useful is Ruby bundler/ Gemfile.

Through https://github.com/coala/coala-quickstart/ I am familiar with  https://gitlab.com/balasankarc/gemfileparser / https://github.com/balasankarc/gemfileparser for Gemfile parsing.

**Comment by DanielVenturini** [comment]
Hi @jayvdb. I'm sorry for the delay, Github didn't send me an email.
This is really a great idea. I never thought of that before, but it's very useful.

Do you think of something like that?:
```bash
lamp --language file
```
For example:
```bash
lamp --ruby path/to/Gemfile
lamp --js path/to/package.json
```

Thanks for the issue.

**Comment by jayvdb** [comment]
IMO the language can usually be detected by the filename, so shorthand cli syntax for the language isnt worth it.  `--language=js` would suffice.  Also many languages have multiple formats for their requirement files.  e.g. requirements.txt, Pipfile, pyproject.toml, etc.  Gemfile, foo.gemspec, etc , so the "language" might actually be better as "format".

I strongly recommend using @dephell , especially for Python because it handles all of the Python file types.  But it also handles non-Python requirement types. c.f. https://github.com/dephell/dephell_specifier/issues/3

So this library could have custom parsers for the file structures were necessary and then have a common class system for the important data.  And also the maintainer(s) of dephell has been very helpful, and they dont mind the scope expanding beyond Python even if they do not intend to focus on non-Python needs at the moment.

I raised https://github.com/balasankarc/gemfileparser/issues/6 for Gemfile parsing related issues.

**Comment by DanielVenturini** [comment]
>IMO the language can usually be detected by the filename [...] so the "language" might actually be better as "format".

Yes, it's true. I hadn't think about it, but it's really the better way to detect the language.

>I strongly recommend using @dephell

I saw about this and it's the better way to check the ranges specifications to `package.json` and `Gemfile`.

Do you know how to get the release times from the Gem API?

**Comment by jayvdb** [comment]
I haven't investigated the Gem API much.

I do know how to obtain that information from the PyPI clients, at least using the google bigdataset.  I did a survey of clients at https://github.com/hugovk/top-pypi-packages/issues/4#issuecomment-491150988

**Comment by DanielVenturini** [comment]
Ok, @jayvdb thanks for the information. I'll do some of the investigations and start work.
